addRelationship

qarth opened the pull request:
Introduced 42 errors, practising PR.

robertjwright committed:
Added checkChildren, breakRelationship, addRelationship and go fmt.
(32 errors apparently...)

robertjwright committed:
Fixed pesudoCtx check

robertjwright committed:
Fixed typo

qarth commented on pseudo.go:
```diff
@@ -236,12 +293,53 @@ func (n *node) pushExcess() {
 // (*node) checkChildren
+func (n *node) checkChildren(curNode *node) {
+	//FIXME
+	for ; curNode.nextScan; curNode.nextScan = curNode.nextScan.next {
```
Unsure about this, original line: for ( ; (curNode->nextScan); curNode->nextScan = curNode->nextScan->next)

clbanning replied:
should be
`for ; curNode.nextScan != nil; curNode.nextScan = curNode.nextScan.next {`

clbanning replied:
The declaration for pushUpward needs to be:
`func (a *arc) pushUpward (child, parent *node, resCap uint)` 
and all instances of `currentArc` get replaced with `a`.

"stats" are in the struct "stats" so that we can serialize them using encoding/json.  So 
`numPushes++` should be `stats.NumPushes` - members are capitalized so that they can be marshaled with stdlib.

... same for `pushDownward`, `addRelationship`, and `checkChildren`

clbanning commented:
Just getting back to this - been busy getting a demo ready.  Is there more coming? Or would you like me to pull this and then clean it up?  Then you could resync to master.

robertjwright committed:
Fixed typo in statistics
Fixed checkChildren, pushUpward, pushDownward, addRelationship, stats.NumPushes

robertjwright committed:
Added findWeakNode

robertjwright committed:
Fixed typo

robertjwright committed:
Fixed typo, removed a FIXME

qarth commented:
Sorry, been busy with graduate applications and psychometric testing. I fixed typos and added findWeakNode, maybe I should have checked out a new branch first because it's a new feature? 
If you could pull this and clean it up that would be great. 

There are a few unresolved references to findWeakNode, strongNode.nextArc, n and strongRoot. There are also a few discrepancies between types uint and int. Should we just be consistent and use all ints? Are there any benefits to using unsigned integers?

robertjwright committed:
Fixed typo

clbanning commented:
We should probably be consistent with original
      and use "unit".  I suspect this may harken back to older 16-bit
      and 32-bit architectures where you might be working with "large"
      data sets.  Today on 64-bit architectures, I suspect "int" would
      be fine; but there's nothing wrong with "uint".
    
    On 4/1/17 4:44 AM, qarth wrote:
    
    
      Sorry, been busy with graduate applications and psychometric
        testing. I fixed typos and added findWeakNode, maybe I should
        have checked out a new branch first because it's a new feature?
        If you could pull this and clean it up that would be great.
      There are a few unresolved references to findWeakNode,
        strongNode.nextArc, n and strongRoot. There are also a few
        discrepancies between types uint and int. Should we just be
        consistent and use all ints? Are there any benefits to using
        unsigned integers?
      —
        You are receiving this because you commented.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/clbanning/pseudo","title":"clbanning/pseudo","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/clbanning/pseudo"}},"updates":{"snippets":[{"icon":"PERSON","message":"@qarth in #1: Sorry, been busy with graduate applications and psychometric testing. I fixed typos and added findWeakNode, maybe I should have checked out a new branch first because it's a new feature? \r\nIf you could pull this and clean it up that would be great. \r\n\r\nThere are a few unresolved references to findWeakNode, strongNode.nextArc, n and strongRoot. There are also a few discrepancies between types uint and int. Should we just be consistent and use all ints? Are there any benefits to using unsigned integers? "}],"action":{"name":"View Pull Request","url":"https://github.com/clbanning/pseudo/pull/1#issuecomment-290912210"}}}

robertjwright committed:
Typo's and &strongRoots - 9 errors remaining
